### PR TITLE
Break SectorAccessor out from RetrievalProviderNode

### PR DESCRIPTION
**Pull request by dirkmc**

### CODE REVIEW

File: retrievalmarket/impl/provider.go [hunk 1]
```diff
@@ -385,6 +389,66 @@ func (p *Provider) HandleQueryStream(stream rmnet.RetrievalQueryStream) {
 	sendResp(answer)
 }
 
+func (p *Provider) getPieceInfoFromCid(ctx context.Context, payloadCID, pieceCID cid.Cid) (piecestore.PieceInfo, bool, error) {
```
Comment (dirkmc) [inline retrievalmarket/impl/provider.go:392]: Moved this from provider_environments.go

File: retrievalmarket/impl/provider_environments.go [hunk 2]
```diff
@@ -178,21 +178,6 @@ func (pde *providerDealEnvironment) DeleteStore(dealID retrievalmarket.DealID) e
 	return nil
 }
 
-func pieceInUnsealedSector(ctx context.Context, n retrievalmarket.RetrievalProviderNode, pieceInfo piecestore.PieceInfo) bool {
```
Comment (dirkmc) [inline retrievalmarket/impl/provider_environments.go:181]: Moved this to provider.go

File: retrievalmarket/impl/provider_environments.go [hunk 3]
```diff
@@ -248,51 +233,6 @@ func getAllDealsContainingPayload(pieceStore piecestore.PieceStore, payloadCID c
 	return dealsIds, nil
 }
 
-func getPieceInfoFromCid(ctx context.Context, n retrievalmarket.RetrievalProviderNode, pieceStore piecestore.PieceStore, payloadCID, pieceCID cid.Cid) (piecestore.PieceInfo, bool, error) {
```
Comment (dirkmc) [inline retrievalmarket/impl/provider_environments.go:251]: Moved this to provider.go

File: retrievalmarket/impl/testnodes/sectoraccessor.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+)
+
+type sectorKey struct {
+	sectorID abi.SectorNumber
+	offset   abi.UnpaddedPieceSize
+	length   abi.UnpaddedPieceSize
+}
+
+// TestSectorAccessor is a mock implementation of the SectorAccessor
+type TestSectorAccessor struct {
```
Comment (dirkmc) [inline retrievalmarket/impl/testnodes/sectoraccessor.go:26]: Just extracted the sealing pieces from test_retrieval_provider_node.go

File: sectoraccessor/sectoraccessor.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package sectoraccessor
```
Comment (raulk) [inline sectoraccessor/sectoraccessor.go:1]: Do we _need_ the new package? Seems like all usages are from `retrievalmarket`, so this interface could just live there? (really trying to curtail the proliferation of micro-packages)

Comment (dirkmc) [reply sectoraccessor/sectoraccessor.go:1]: I went back and forth on it - we also use this package from the miner API so it's not strictly to do with retrieval market. Fine with me to put it in the retrieval market package though.

**Pull request merged**